WeatherMonitoringAndReportingService.Tests/DataSourceProcessor/JsonFileProcessorTests.cs

```diff
@@ -0,0 +1,98 @@
+﻿using AutoFixture;
+using Moq;
+using WeatherMonitoringAndReportingService.AppSettings;
+using WeatherMonitoringAndReportingService.Config;
+using WeatherMonitoringAndReportingService.DataSourceProcessor;
+using WeatherMonitoringAndReportingService.DataSourceProcessor.Readers;
+using WeatherMonitoringAndReportingService.DataSourceProcessor.Serializers;
+using WeatherMonitoringAndReportingService.DataSourceProcessor.Writers;
+
+namespace WeatherMonitoringAndReportingService.Tests.DataSourceProcessor;
+
+public class JsonFileProcessorTests
+{
+    private readonly Mock<IFileWriter> _jsonWriterMock;
+    private readonly Mock<IFileSerializer> _jsonSerializerMock;
+    private readonly Mock<IFileReader> _jsonReaderMock;
+    private readonly JSONFileProcessor _jsonProcessor;
+    private readonly Fixture _fixture;
+
+    public JsonFileProcessorTests()
+    {
+        _fixture = new Fixture();
+        _jsonWriterMock = new Mock<IFileWriter>();
+        _jsonSerializerMock = new Mock<IFileSerializer>();
+        _jsonReaderMock = new Mock<IFileReader>();
+        _jsonProcessor = new JSONFileProcessor(_jsonWriterMock.Object, _jsonSerializerMock.Object, _jsonReaderMock.Object);
+
+        _jsonSerializerMock
```

Comment:
Please move the below setup lines into a function named SetupCommonMocks() and call it where needed.
Currently you have added a general mocking behaviour inside the constructor which usually we avoid it.